.github/workflows/run_tests.yml

```diff
@@ -121,79 +112,87 @@ jobs:
-
-      - name: Overwrite redis cluster env vars
+          SCRATCH_DIR="single_scratch"
+          rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
+          redis-server --port 6380 --daemonize yes \
+                       --logfile "single.log" \
+                       --loadmodule $REDISAI_MODULES &&
+          sleep 5; cat single.log &&
+          popd &&
+          export SSDB=$SSDB_SINGLE SMARTREDIS_TEST_CLUSTER=False &&
+          make test-verbose COV_FLAGS="--cov=./src/python/module/smartredis/ --cov-report=xml --cov-append"
+
+      - name: Clustered DB tests
         run: |
-         echo "SSDB=127.0.0.1:7000,127.0.0.1:7001,127.0.0.1:7002" >> $GITHUB_ENV &&
-         echo "SMARTREDIS_TEST_CLUSTER=True" >> $GITHUB_ENV
-
-      - name: Start redis cluster
-        run: redis-cli --cluster create $(echo $SSDB | tr "," " ") --cluster-yes
-
-      # CI tests that should be run with clustered Redis go here
-      - name: Run testing with redis cluster
+          SCRATCH_DIR="cluster_scratch"
+          rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
+          for port in {7000..7002}; do
+            redis-server --port $port --cluster-enabled yes --daemonize yes \
+                         --cluster-config-file "$port.conf" --protected-mode no --save "" \
+                         --logfile "$port.log" \
+                         --loadmodule $REDISAI_MODULES
+          done &&
+          sleep 2 &&
+          redis-cli --cluster create $(echo $SSDB_CLUSTERED | tr "," " ") --cluster-yes &&
+          export SSDB=$SSDB_CLUSTERED SMARTREDIS_TEST_CLUSTER=True &&
+          popd &&
+          python -m pytest --cov=./src/python/module/smartredis/ --cov-report=xml --cov-append \
+                 --ignore ./tests/docker -vv -s ./tests
+
+      - name: UDS DB tests
         run: |
-          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/install/lib &&
-          export SR_LOG_FILE=smartredis_cicd_tests_log.txt &&
-          export SR_LOG_LEVEL=INFO &&
-          make test-verbose
-
+          mkdir -p /tmp && ## Create socketfile for UDS testing
+          touch /tmp/redis.sock &&
+          chmod 777 /tmp/redis.sock &&
+          SCRATCH_DIR="uds_scratch" &&
+          rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
+          redis-server --unixsocket /tmp/redis.sock --unixsocketperm 777 --port 0 --bind 127.0.0.1 \
+                       --daemonize yes --protected-mode no --logfile "uds.log" \
+                       --loadmodule $REDISAI_MODULES &&
+          popd &&
+          export SSDB=$SSDB_UDS SMARTREDIS_TEST_CLUSTER=False &&
+          python -m pytest --cov=./src/python/module/smartredis/ --cov-report=xml --cov-append \
+                 --ignore ./tests/docker -vv -s ./tests
+
+      - name: Collect C++ coverage
+        run: third-party/lcov/install/usr/local/bin/lcov -c -d build/CMakeFiles -d tests -o coverage.info
+
+      # Upload final coverage reports
```

Comment:
Suggest changing this comment to "Collect and upload..." and moving it to line 191 to logically group the last two steps together

Reply:
Also, is this still just C++ coverage?

Reply:
Thank you. It is all the compiled language tests.